### ISSUE TITLE
`merge` in utils convert an array post data to an object

### ISSUE THREAD

**Issue by gejiawen** [axios/axios#1641]
#### Summary

try

```js
let url = '/dsk',
let data = {
    dsk: ['a1', 'a2']
}
let config = {
    baseURL: 'http://8.8.8.8',
    data
}

axios.post(url, data, {config}).then(resp => { console.log(resp.config.data) })
```

get `resp.config.data`

```
{"baseURL": "http://8.8.8.8", "url": "/dsk", "data": {"0": "a1", "1": "a2"}}
```

this config data is not expect. 

#### Context

- axios version: v0.18.0
- Environment: node v10.4.1

**Comment by gaofang** [comment]
I have recently met this problem,I looked at axios's source code and found the problem.
FInd function 'merge' in util.js, and you can find this mege function has defect for Array.
Make the following change and your problem will be solved.
`function merge(/* obj1, obj2, obj3, ... */) {
  var result = {};
  function assignValue(val, key) {
    if (typeof result[key] === 'object' && typeof val === 'object') {

        /* add this code for array */
        if (isArray(result[key])) {
            result[key].push.apply(result[key], val);
            return
        }
        /* add this code for array */

      result[key] = merge(result[key], val);
    } else {
      result[key] = val;
    }
  }

  for (var i = 0, l = arguments.length; i < l; i++) {
    forEach(arguments[i], assignValue);
  }
  return result;
}`

**Comment by seedlinktech-william** [comment]
I have met same situation.  What I used for workaround is delete the `data` in config, like:
```
let url = '/dsk',
let data = {
    dsk: ['a1', 'a2']
}
let config = {
    baseURL: 'http://8.8.8.8',
    // data
}

axios.post(url, data, config).then(resp => { console.log(resp.config.data) })
``` 
When the merge be invoked, it will check the arguments of `axios.post` and compare the `data` with `config.data`.  If there `config.data` is false value, it will using the `data` directly.  Then the `merge` will not be called.

**Comment by xz491839007** [comment]
I have met same situation

**Comment by ronkorving** [comment]
@gaofang Can you make PR to Axios with your fix?

**Comment by sjcl** [comment]
No solution for DELETE request.

**Comment by chinesedfan** [comment]
Closed in favor of #2567.
